Error starting provider in barge: ModuleNotFoundError: No module named 'py'

# Error

```console
provider2_1        | [2021-08-05 14:28:38 +0000] [18] [ERROR] Exception in worker process
provider2_1        | Traceback (most recent call last):
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/arbiter.py", line 583, in spawn_worker
provider2_1        |     worker.init_process()
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/workers/base.py", line 119, in init_process
provider2_1        |     self.load_wsgi()
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/workers/base.py", line 144, in load_wsgi
provider2_1        |     self.wsgi = self.app.wsgi()
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/base.py", line 67, in wsgi
provider2_1        |     self.callable = self.load()
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/wsgiapp.py", line 49, in load
provider2_1        |     return self.load_wsgiapp()
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/app/wsgiapp.py", line 39, in load_wsgiapp
provider2_1        |     return util.import_app(self.app_uri)
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/gunicorn/util.py", line 358, in import_app
provider2_1        |     mod = importlib.import_module(module)
provider2_1        |   File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
provider2_1        |     return _bootstrap._gcd_import(name[level:], package, level)
provider2_1        |   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
provider2_1        |   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
provider2_1        |   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
provider2_1        |   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
provider2_1        |   File "<frozen importlib._bootstrap_external>", line 783, in exec_module
provider2_1        |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
provider2_1        |   File "/ocean-provider/ocean_provider/run.py", line 15, in <module>
provider2_1        |     from ocean_provider.myapp import app
provider2_1        |   File "/ocean-provider/ocean_provider/myapp.py", line 18, in <module>
provider2_1        |     from .database import Base, SessionLocal, engine
provider2_1        |   File "/ocean-provider/ocean_provider/database.py", line 11, in <module>
provider2_1        |     from ocean_provider.utils.basics import get_config
provider2_1        |   File "/ocean-provider/ocean_provider/utils/basics.py", line 10, in <module>
provider2_1        |     from ocean_lib.common.aquarius.aquarius import Aquarius
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/ocean_lib/common/aquarius/__init__.py", line 10, in <module>
provider2_1        |     from .aquarius_provider import AquariusProvider  # noqa
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/ocean_lib/common/aquarius/aquarius_provider.py", line 9, in <module>
provider2_1        |     from .aquarius import Aquarius
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/ocean_lib/common/aquarius/aquarius.py", line 15, in <module>
provider2_1        |     from ocean_lib.common.ddo.ddo import DDO
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/ocean_lib/common/ddo/ddo.py", line 22, in <module>
provider2_1        |     from ocean_lib.data_provider.data_service_provider import DataServiceProvider
provider2_1        |   File "/usr/local/lib/python3.8/dist-packages/ocean_lib/data_provider/data_service_provider.py", line 27, in <module>
provider2_1        |     from py._path.local import LocalPath
provider2_1        | ModuleNotFoundError: No module named 'py'
```

# Analysis

This is the function where `LocalPath` is used.

```python
@staticmethod
    def write_file(
        response: Response, destination_folder: Union[LocalPath, str], file_name: str
    ) -> None:
        """
        Write the response content in a file in the destination folder.
        :param response: Response
        :param destination_folder: Destination folder, string
        :param file_name: File name, string
        :return: None
        """
        if response.status_code == 200:
            with open(os.path.join(destination_folder, file_name), "wb") as f:
                for chunk in response.iter_content(chunk_size=None):
                    f.write(chunk)
            logger.info(f"Saved downloaded file in {f.name}")
        else:
            logger.warning(f"consume failed: {response.reason}")
```

It's calling `os.path.join`. According to the [docs]:(https://docs.python.org/3/library/os.path.html#os.path.join)

> Changed in version 3.6: Accepts a [path-like object](https://docs.python.org/3/glossary.html#term-path-like-object) for path and paths.

>path-like object
>An object representing a file system path. A path-like object is either a str or bytes object representing a path, or an object implementing the os.PathLike protocol.

Thus, the `write_file` type hint should be
`Union[str, bytes, os.PathLike]`

--- a/ocean_lib/data_provider/data_service_provider.py
+++ b/ocean_lib/data_provider/data_service_provider.py
@@ -24,7 +24,6 @@ from ocean_lib.models.algorithm_metadata import AlgorithmMetadata
 from ocean_lib.ocean.env_constants import ENV_PROVIDER_API_VERSION
 from ocean_lib.web3_internal.transactions import sign_hash
 from ocean_lib.web3_internal.wallet import Wallet
-from py._path.local import LocalPath
 from requests.exceptions import InvalidURL
 from requests.models import PreparedRequest, Response
 from requests.sessions import Session
@@ -530,7 +529,9 @@ class DataServiceProvider:
 
     @staticmethod
     def write_file(
-        response: Response, destination_folder: Union[LocalPath, str], file_name: str
+        response: Response,
+        destination_folder: Union[str, bytes, os.PathLike],
+        file_name: str,
     ) -> None:
         """
         Write the response content in a file in the destination folder.

